src/components/snackbar/snackbar.tsx

```diff
@@ -99,89 +102,92 @@ export class Snackbar {
     @Event()
     private hide: EventEmitter<void>;
 
-    private mdcSnackbar: MDCSnackbar;
+    @State()
+    private offset: number = 0;
 
-    constructor() {
-        this.handleMdcClosing = this.handleMdcClosing.bind(this);
-    }
+    @State()
+    private open: boolean = false;
 
-    public connectedCallback() {
-        this.initialize();
-    }
+    @State()
+    private closing: boolean = true;
 
-    public componentDidLoad() {
-        this.initialize();
-    }
-
-    private initialize() {
-        const element = this.host.shadowRoot.querySelector('.mdc-snackbar');
-        if (!element) {
-            return;
-        }
-
-        this.mdcSnackbar = new MDCSnackbar(element);
-
-        this.mdcSnackbar.listen('MDCSnackbar:closing', this.handleMdcClosing);
-    }
-
-    public disconnectedCallback() {
-        this.mdcSnackbar.unlisten('MDCSnackbar:closing', this.handleMdcClosing);
-        this.mdcSnackbar.destroy();
+    @Listen('changeOffset')
+    protected onChangeIndex(event: CustomEvent<number>) {
+        event.stopPropagation();
+        this.offset = event.detail;
     }
 
     /**
      * Show the snackbar
      */
     @Method()
     public async show() {
-        if (this.timeout) {
-            this.mdcSnackbar.timeoutMs = this.timeout;
+        if (this.open) {
+            return;
         }
 
-        this.mdcSnackbar.labelText = this.message;
+        this.open = true;
+        this.closing = false;
+        container.add(this.host);
 
-        this.mdcSnackbar.open();
+        if (this.timeout) {
+            setTimeout(
+                this.close,
+                Math.max(this.timeout - hideAnimationDuration, 0),
+            );
+        }
     }
 
+    private close = () => {
+        if (!this.open) {
+            return false;
+        }
+
+        this.closing = true;
+
+        setTimeout(() => {
+            this.open = false;
+            container.remove(this.host);
+            this.hide.emit();
+            this.offset = 0;
+        }, hideAnimationDuration);
+    };
```

Comment:
These timeouts can conflict. If you open a snackbar in the docs, then click the button to close it, then open it again somewhat quickly, the first closing timeout is still active, and will close the snackbar really quickly after it opens the second time.

If we store the timeout handle from the `show` method, we can cancel that timeout in the `close` function. Then we'd avoid this problem.

I'm not sure if there can be some broken interaction if the `show` method is called during the timeout set in the `close` function as well, but it might be best to make sure any active timeout gets cancelled in that case too.

Reply:
d776e6e47